app/helpers/ExerciseConfig/Pipeline/Box/Boxes/JvmCompilationBox.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace App\Helpers\ExerciseConfig\Pipeline\Box;
+
+use App\Exceptions\ExerciseConfigException;
+use App\Helpers\ExerciseConfig\Compilation\CompilationParams;
+use App\Helpers\ExerciseConfig\Pipeline\Box\Params\ConfigParams;
+use App\Helpers\ExerciseConfig\Pipeline\Box\Params\Priorities;
+use App\Helpers\ExerciseConfig\Pipeline\Box\Params\TaskCommands;
+use App\Helpers\ExerciseConfig\Pipeline\Box\Params\TaskType;
+use App\Helpers\ExerciseConfig\Pipeline\Ports\Port;
+use App\Helpers\ExerciseConfig\Pipeline\Ports\PortMeta;
+use App\Helpers\ExerciseConfig\VariableTypes;
+use App\Helpers\JobConfig\Tasks\Task;
+use Nette\Utils\Strings;
+
+/**
+ * Box which represents custom compilation to JVM bytecode.
+ */
+class JvmCompilationBox extends CompilationBox
+{
+    /** Type key */
+    public static $BOX_TYPE = "jvm-compilation";
+    public static $COMPILATION_SUBDIR = 'compiled-classes';
+    public static $CLASS_FILES_WILDCARD = "*.class";
+    public static $CLASS_FILES_PORT_KEY = "class-files";
+    public static $JAR_FILES_PORT_KEY = "jar-files";
+    public static $COMPILER_EXEC_PORT_KEY = "compiler-exec";
+    public static $DEFAULT_NAME = "JVM Custom Compilation";
+
+    private static $initialized = false;
+    private static $defaultInputPorts;
+    private static $defaultOutputPorts;
+
+    /**
+     * Static initializer.
+     * @throws ExerciseConfigException
+     */
+    public static function init()
+    {
+        if (!self::$initialized) {
+            self::$initialized = true;
+            self::$defaultInputPorts = array(
+                new Port((new PortMeta())->setName(self::$ARGS_PORT_KEY)->setType(VariableTypes::$STRING_ARRAY_TYPE)),
+                new Port(
+                    (new PortMeta())->setName(self::$SOURCE_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                ),
+                new Port(
+                    (new PortMeta())->setName(self::$EXTRA_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                ),
+                new Port((new PortMeta())->setName(self::$JAR_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)),
+                new Port((new PortMeta())->setName(self::$COMPILER_EXEC_PORT_KEY)->setType(VariableTypes::$STRING_TYPE))
+            );
+            self::$defaultOutputPorts = array(
+                new Port(
+                    (new PortMeta())->setName(self::$CLASS_FILES_PORT_KEY)->setType(VariableTypes::$FILE_ARRAY_TYPE)
+                )
+            );
+        }
+    }
+
+    /**
+     * JudgeNormalBox constructor.
```

Comment:
I do not want to judge, but it seems like Ctrl+C/Ctrl+V issue (which is a poor way how to pay tribute to late Larry Tesler...)